NodeJS/src/AgentClient.ts

```diff
@@ -144,11 +165,47 @@ export class AgentClient {
 
     }
 
+    public fetchDefaultSkillNumber(): Promise<number|undefined> {
+        return this.restClient.getSubAccountAgentSkills(this.subAccountId)
+            .then((response: { data: { [x: string]: { [x: string]: any } } }) => {
+                logger.debug(response.data)
+                // data has a key of skillResponse, which is a map of subaccountId, and array of skillResponse
+                // skillResponses is SkillResponse[]
+                let skillResponses = response.data['skillResponses'][this.subAccountId];
+                if (skillResponses) {
+                    let defaultSkill = skillResponses.find((skillResponse: { number: number, name: string }) => skillResponse.name === Constants.DEFAULT_SKILL_NAME);
+                    return !!defaultSkill ? defaultSkill.number : undefined
```

Comment:
Assuming that you are up to latest, you can use [Optional Chaining](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#optional-chaining) to simplify this to: `defaultSkill?.number`

Reply:
Have seen it used by others.  Time to make it first instinct.